projects/storefrontlib/src/cms-structure/page/component/handlers/web-component.handler.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { TestBed } from '@angular/core/testing';
+import { WebComponentHandler } from './web-component.handler';
+import { CmsMappingService } from '../../../services/cms-mapping.service';
+import { Component, ViewContainerRef } from '@angular/core';
+import { Priority } from '@spartacus/core';
+import { take } from 'rxjs/operators';
+import { CxApiService } from '../services/cx-api.service';
+
+const mockCmsMappingService = jasmine.createSpyObj('CmsMappingService', [
+  'getComponentMapping',
+]);
+
+@Component({
+  template: '',
+})
+class WrapperComponent {
+  constructor(public vcr: ViewContainerRef) {}
+}
+
+describe('WebComponentHandler', () => {
+  let handler: WebComponentHandler;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        {
+          provide: CmsMappingService,
+          useValue: mockCmsMappingService,
+        },
+        {
+          provide: CxApiService,
+          useValue: {},
+        },
+      ],
+    });
+    handler = TestBed.inject(WebComponentHandler);
+  });
+
+  it('should be created', () => {
+    expect(handler).toBeTruthy();
+  });
+
+  describe('hasMatch', () => {
+    it('should match component class', () => {
+      expect(handler.hasMatch({ component: '#my-webcomponent' })).toBeTruthy();
+    });
+  });
+
+  it('getPriority should return low', () => {
+    expect(handler.getPriority()).toEqual(Priority.LOW);
+  });
+
+  it('should launch component', (done) => {
+    const fixture = TestBed.createComponent(WrapperComponent);
+    fixture.detectChanges();
+
+    handler
+      .launcher(
+        { component: '#my-webcomponent' },
+        fixture.componentInstance.vcr
+      )
+      .pipe(take(1))
+      .subscribe(([elRef]) => {
+        expect(elRef.nativeElement.tagName).toBe('MY-WEBCOMPONENT');
+        done();
```

Comment:
P2: Can we also verify that the web component was appended inside the WrapperComponent?